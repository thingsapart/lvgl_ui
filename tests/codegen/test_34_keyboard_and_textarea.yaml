# A keyboard that automatically targets a text area.
- type: obj
  layout: LV_LAYOUT_FLEX
  flex_flow: LV_FLEX_FLOW_COLUMN
  size: [300, 250]
  center: []
  children:
    - type: textarea
      id: '@ta'
      width: 100%
      one_line: false
    - type: keyboard
      width: 100%
      flex_grow: 1
      textarea: '@ta'
