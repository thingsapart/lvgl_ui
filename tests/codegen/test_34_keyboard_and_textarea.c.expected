/* AUTO-GENERATED by the 'c_code' backend */

#include "lvgl.h"
#include "c_gen/lvgl_dispatch.h" // For obj_registry_add
#include "data_binding.h"

void create_ui(lv_obj_t* parent) {
    // unnamed: obj_0 (obj)
    lv_obj_t* obj_0 = lv_obj_create(parent);

    lv_obj_set_style_layout(obj_0, LV_LAYOUT_FLEX, 0);
    lv_obj_set_style_flex_flow(obj_0, LV_FLEX_FLOW_COLUMN, 0);
    lv_obj_set_style_size(obj_0, 300, 250, 0);
    lv_obj_center(obj_0);
    // ta: ta_1 (textarea)
    lv_obj_t* ta_1 = lv_textarea_create(obj_0);

    obj_registry_add("ta", ta_1);
    lv_obj_set_style_width(ta_1, lv_pct(100), 0);
    lv_textarea_set_one_line(ta_1, false);

    // unnamed: keyboard_2 (keyboard)
    lv_obj_t* keyboard_2 = lv_keyboard_create(obj_0);

    lv_obj_set_style_width(keyboard_2, lv_pct(100), 0);
    lv_obj_set_style_flex_grow(keyboard_2, 1, 0);
    lv_keyboard_set_textarea(keyboard_2, ta_1);


}
